tests/src/tests/defuse/storage/mod.rs

```diff
@@ -0,0 +1,232 @@
+    None
+)]
+#[trace]
+#[case(
+    MIN_FT_STORAGE_DEPOSIT_VALUE.checked_add(ONE_YOCTO_NEAR).unwrap(),
+    Some(MIN_FT_STORAGE_DEPOSIT_VALUE)
+)]
+async fn storage_deposit_success(
+    random_seed: Seed,
+    #[case] amount_to_deposit: NearToken,
+    #[case] expected_deposited: Option<NearToken>,
+) {
+    let mut rng = make_seedable_rng(random_seed);
+
+    let env = Env::builder().disable_ft_storage_deposit().build().await;
+
+    env.fund_account_with_near(env.user1.id(), NearToken::from_near(1000))
+        .await;
+    env.fund_account_with_near(env.user2.id(), NearToken::from_near(1000))
+        .await;
+    env.fund_account_with_near(env.defuse.id(), NearToken::from_near(10000))
+        .await;
+
+    {
+        let storage_balance_ft1_user1 = env
+            .storage_balance_of(&env.ft1, env.user1.id())
+            .await
+            .unwrap();
+
+        let storage_balance_ft1_user2 = env
+            .storage_balance_of(&env.ft1, env.user2.id())
+            .await
+            .unwrap();
+
+        assert!(storage_balance_ft1_user1.is_none());
+        assert!(storage_balance_ft1_user2.is_none());
+    }
+
+    // For intents contract to have a balance in wnear, we make a storage deposit for it
+    env.storage_deposit(
+        env.wnear.id(),
+        Some(env.defuse.id()),
+        NearToken::from_near(1),
+    )
+    .await
+    .unwrap();
+
+    env.storage_deposit(&env.ft1, Some(env.user1.id()), NearToken::from_near(1))
+        .await
+        .unwrap();
+
+    {
+        let storage_balance_ft1_user1 = env
+            .storage_balance_of(&env.ft1, env.user1.id())
+            .await
+            .unwrap();
+
+        let storage_balance_ft1_user2 = env
+            .storage_balance_of(&env.ft1, env.user2.id())
+            .await
+            .unwrap();
+
+        assert_eq!(
+            storage_balance_ft1_user1.unwrap().total,
+            MIN_FT_STORAGE_DEPOSIT_VALUE
+        );
+        assert!(storage_balance_ft1_user2.is_none());
+    }
+
+    // The user should have enough wnear in his account (in his account in the wnear contract)
+    env.user2
+        .near_deposit(env.wnear.id(), NearToken::from_near(100))
+        .await
+        .unwrap();
+
+    // Fund the user's account with near in the intents contract for the storage deposit intent
+    env.defuse_ft_mint(
+        env.wnear.id(),
+        NearToken::from_near(10).as_yoctonear(),
+        env.user2.id(),
+    )
+    .await
+    .unwrap();
+
+    let storage_deposit_intent = StorageDeposit {
+        contract_id: env.ft1.clone(),
+        account_id: env.user2.id().clone(),
+        amount: amount_to_deposit,
+    }
+    .into();
+
+    let intents = DefuseIntents {
+        intents: [storage_deposit_intent].into(),
+    };
+
+    let signed_intents = env.user2.sign_defuse_message(
+        env.defuse.id(),
+        rng.random(),
+        Deadline::timeout(std::time::Duration::from_secs(120)),
+        intents,
```

Comment:
It's hard to read when these values are spread over the code. Inlining them is more readable:
```suggestion
        DefuseIntents {
           intents: [StorageDeposit {
               contract_id: env.ft1.clone(),
               account_id: env.user2.id().clone(),
               amount: amount_to_deposit,
           }.into()]
         .into(),
       }
```

Reply:
Actually I have the exact opposite view... cramming code makes it harder for me to read. In fact, this was difficult for me when I first started reading the code. 

If you insist, I'd do it, but I'm not particularly happy about it.

One thing I can suggest is wrapping these in one scope, like I did above. This way, it prevents mixing them with other, unrelated things.

Reply:
My reasoning for inlining here is that it should be as close as possible to JSON representation of the signed intent, that will be submitted on-chain, e.g.:
```json
{
  "signed": [
    {
      "payload": "{\"signer_id\":\"465c5516b703bc8a925f8e0f4a47337f69302be899050f2904bd5dce5f07afde\",\"verifying_contract\":\"intents.near\",\"deadline\":\"2025-03-20T09:54:34.684Z\",\"nonce\":\"4dusNOOVIVXuB7z6F8LAOCKXfEGvMBiYm5nlH4ltTg0=\",\"intents\":[{\"intent\":\"ft_withdraw\",\"token\":\"sol-5ce3bf3a31af18be40ba30f721101b4341690186.omft.near\",\"receiver_id\":\"sol-5ce3bf3a31af18be40ba30f721101b4341690186.omft.near\",\"amount\":\"4000000\",\"memo\":\"WITHDRAW_TO:5jfCVJ7oZ2Y38Nu28qCdRuY5KgsuqRKZtwYzCY7q3DNV\"}]}",
      "public_key": "ed25519:5jfCVJ7oZ2Y38Nu28qCdRuY5KgsuqRKZtwYzCY7q3DNV",
      "signature": "ed25519:3udGxDpiVwSjA7bNJbCfqZ7S72HffvSvRY9vG9tEcbZbinNotRfipw65AXUpCtq4FyxHvpyKuk1cFynQsXTnPKx5",
      "standard": "raw_ed25519"
    }
  ]
}
```

Reply:
I don't love it... but alright. Done.